pkgs/by-name/ed/edopro/package.nix

```diff
@@ -244,21 +317,62 @@ let
         "textures"
         "WindBot"
       ];
-    in
-    writeShellScriptBin "edopro" ''
-      set -eu
-      EDOPRO_DIR="''${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
+      wrapperZenityMessageTemplate = writeText "edopro-wrapper-multiple-versions-message.txt.in" ''
+        Nixpkgs' EDOPro wrapper has found more than 1 directory in: ''${EDOPRO_BASE_DIR}
 
-      if [ ! -d $EDOPRO_DIR ]; then
-          mkdir -p $EDOPRO_DIR
-          cp -r --no-preserve=all ${assets}/{${assetsToCopy}} $EDOPRO_DIR
-          chmod -R go-rwx $EDOPRO_DIR
+        We expected the only directory to be: ''${EDOPRO_DIR}
 
-          rm $EDOPRO_DIR/config/io.github.edo9300.EDOPro.desktop.in
-      fi
+        There may have been an update, requiring you to migrate any files you care about from an older version.
 
-      exec ${lib.getExe edopro} -C $EDOPRO_DIR $@
-    '';
+        Examples include:
+
+        - decks/*
+        - config/system.conf - which has your client's settings
+        - any custom things you may have installed into: fonts, skins, script, sound, ...
+        - anything you wish to preserve from: replay, screenshots
+
+        Once you have copied over everything important to ''${EDOPRO_DIR}, delete the old version's path.
+      '';
+    in
+    writeShellApplication {
+      name = "edopro";
+      runtimeInputs = [
+        envsubst
+        zenity
+      ];
+      text = ''
+        export EDOPRO_VERSION="${deps.edopro-version}"
+        export EDOPRO_BASE_DIR="''${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
+        export EDOPRO_DIR="''${EDOPRO_BASE_DIR}/''${EDOPRO_VERSION}"
+
+        # If versioned directory doesn't exist yet, make it & copy over assets
+        if [ ! -d "$EDOPRO_DIR" ]; then
+            mkdir -p "$EDOPRO_DIR"
+            cp -r --no-preserve=all ${assets}/{${assetsToCopy}} "$EDOPRO_DIR"
+            chmod -R go-rwx "$EDOPRO_DIR"
+
+            rm "$EDOPRO_DIR"/config/io.github.edo9300.EDOPro.desktop.in
+        fi
+
+        # Different versions provide different assets. Some are necessary for the game to run properly (configs for
+        # where to get incremental updates from, online servers, card scripting, certificates for communication etc),
+        # and some are optional nice-haves (example decks). It's also possible to override assets with custom skins.
+        #
+        # Don't try to manage all of this across versions, just inform the user that they may need to migrate their
```

Comment:
Why don't we handle it, why did we give up so soon?

Reply:
Simplicity, being able to *eventually* get *a* version of these bumps in, lack of spoons to fully figure this out any time soon.

If you want to figure out which files we need to keep synced, and the best way of syncing these without overwriting user-touched files, feel free to send a patch for me to apply.